---
title: C6310 | Microsoft Docs
ms.custom: ''
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-devops-test
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C6310
helpviewer_keywords:
- C6310
ms.assetid: e759eb63-883f-4c3e-bf2f-b924ff511405
caps.latest.revision: 15
author: mikeblome
ms.author: mblome
manager: ghogen
ms.openlocfilehash: 8bbe62582490f45c5bc1a5a7eb6a2c59820a4074
ms.sourcegitcommit: af428c7ccd007e668ec0dd8697c88fc5d8bca1e2
ms.translationtype: MT
ms.contentlocale: es-ES
ms.lasthandoff: 11/16/2018
ms.locfileid: "51800749"
---
# <a name="c6310"></a>C6310
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

Advertencia C6310: una constante no válida en el filtro de excepción puede provocar un comportamiento inesperado  
  
 Este mensaje indica que se ha detectado una constante no válida en la expresión de filtro de controlador de excepciones estructurado. Las constantes definidas para su uso en la expresión de filtro de controlador de excepciones estructurado son:  
  
- `EXCEPTION_CONTINUE_EXECUTION`  
  
- `EXCEPTION_CONTINUE_SEARCH`  
  
- `EXCEPTION_EXECUTE_HANDLER`  
  
  Estos valores se definen en el excpt.h de archivo de encabezado en tiempo de ejecución.  
  
  Uso de una constante que no está en la lista anterior puede provocar un comportamiento inesperado.  
  
## <a name="example"></a>Ejemplo  
 El código siguiente genera esta advertencia:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (EXCEPTION_ACCESS_VIOLATION)  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 Para corregir esta advertencia, utilice el código siguiente:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)  
  {  
    puts("Exception Occurred");  
  }  
}  
```



